### PR TITLE
Scala3doc: Add build step for FE resources

### PR DESCRIPTION
**Pull request by grzegorz-bielski**
~~Requires https://github.com/lampepfl/dotty/pull/10155 and a rebase~~

### Main change
- Added a build step for `scala3doc` FE resources with a TS and Sass support - this will make the codebase more maintainable in the long run and is a base for subsequent refactors.

### Extras
- refactored the filter bar component to TS
- moved the filter bar's styles to `.scss` without any refactor, for now
- Fixed broken e2e tests
- added CI setup scala3doc e2e tests ~~(but it doesn't seem to be picked by PR CI job...)~~ tested here: https://github.com/grzegorz-bielski/dotty/pull/1/checks?check_run_id=1426973874

### CODE REVIEW

File: scala3doc/resources/dotty_res/scripts/bundle.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
....onGroupSelectionChange(t,e)}}))},n.onFilterVisibilityChange=function(){n.setState((function(t){return{isVisible:!t.isVisible}}))},n.onFilterToggle=function(t,e){n.setState((function(n){return{filter:n.filter.onFilterToggle(t,e)}}))},n.refs={elements:i(".documentableElement"),filterBar:r(".documentableFilter")},n.state={filter:new p("",{},n.refs.elements,!0),isVisible:!1},n.input=new w({onInputChange:n.onInputChange}),n.list=new y({filter:n.state.filter}),n.filterGroup=new m({filter:n.state.filter,onFilterToggle:n.onFilterToggle,onGroupSelectChange:n.onGroupSelectChange,onFilterVisibilityChange:n.onFilterVisibilityChange}),n.render(),n}return O(e,t),e.tryAttaching=function(){r(".documentableFilter")&&new e},e.prototype.render=function(){this.refs.filterBar&&(this.state.isVisible?this.refs.filterBar.classList.add("active"):this.refs.filterBar.classList.remove("active")),this.list.render({filter:this.state.filter}),this.filterGroup.render({filter:this.state.filter,onFilterToggle:this.onFilterToggle,onGroupSelectChange:this.onGroupSelectChange,onFilterVisibilityChange:this.onFilterVisibilityChange})},e}(n);window.addEventListener("DOMContentLoaded",(function(){F.tryAttaching()}))})();
```
Comment (smarter) [inline scala3doc/resources/dotty_res/scripts/bundle.js:1]: Can we avoid committing generated files in the repo? It's fine if the build process involves calling npm, it just means the sbt project needs to call npm itself when needed, this is what we do in the vscode-dotty project: https://github.com/lampepfl/dotty/blob/93f24a09ef22e8c150a14531dbb49e7599a70d2f/project/Build.scala#L1231-L1233

Comment (grzegorz-bielski) [reply scala3doc/resources/dotty_res/scripts/bundle.js]: done!

**Comment by grzegorz-bielski** [comment]
@romanowski you might be interested as well

P.S the `Dotty CI / test_bootstrapped-windows` PR check is failing for some reason

**Comment by grzegorz-bielski** [comment]
Although the fix for the e2e tests and the pipeline might be still useful, the rest is pretty outdated at this point.

**Pull request closed without merging**